gradle-conjure/src/main/java/com/palantir/gradle/conjure/ConjureLocalGenerateGenericTask.java

```diff
@@ -27,7 +27,7 @@
 public abstract class ConjureLocalGenerateGenericTask extends ConjureLocalGenerateTask {
 
     private static final Pattern PATTERN =
-            Pattern.compile("^(.*)-([0-9]+\\.[0-9]+\\.[0-9]+(?:-rc[0-9]+)?(?:-g[a-f0-9]+)?)(?:.conjure)?.json$");
+            Pattern.compile("^(.*)-([0-9]+\\.[0-9]+\\.[0-9]+(?:-rc[0-9]+)?(?:-[1-9][0-9]*-g[a-f0-9]+)?)(?:.conjure)?.json$");
```

Comment:
Might be better to simplify this pattern to just extract the version then run it through `OrderableSlsVersion.safeValueOf` from https://github.com/palantir/sls-version-java. That way we don't have to redefine the version regexes here.